ComponentTextKit/TextKit/CKTextKitContext.h

```diff
@@ -26,7 +26,8 @@
 - (instancetype)initWithAttributedString:(NSAttributedString *)attributedString
                            lineBreakMode:(NSLineBreakMode)lineBreakMode
                     maximumNumberOfLines:(NSUInteger)maximumNumberOfLines
-                         constrainedSize:(CGSize)constrainedSize;
+                         constrainedSize:(CGSize)constrainedSize
+                   layoutManagerSubclass:(Class)layoutManagerSubclass;
```

Comment:
We have to add documentation for what this is and why you might use it.  There must also be a warning in this docblock indicating that whatever layout manager class you provide here (while only accessed from one thread at a time) multiple instances will be in use at the same time on different threads.  Any global caches must be strongly protected.